sync.py

```diff
@@ -30,13 +35,183 @@ def __init__(self):
         self.codes_found_in_cascade = []
         self.missing_data_codes = []
 
-    def process_all_blocks(self, wsapi_data, time_to_wait, send_email_after):
+    def convert_dictionary_to_hash(self, dictionary):
+
+        def _recursively_alphabetize_dictionary_by_keys(dictionary_to_alphabetize):
+            if isinstance(dictionary_to_alphabetize, dict):
+                to_return = '{'
+                ordered_keys = sorted(dictionary_to_alphabetize.keys())
+                for key in ordered_keys:
```

Comment:
did you experiment with using python's build in `str()` method to handle building the string to md5?

Reply:
As in, change `alphabetized_string = _recursively_alphabetize_dictionary_by_keys(dictionary)` to `alphabetized_string = str(dictionary)`?

I have not tried that -- I assumed that since Python can return dictionary keys in any order that the built-in str() converter would not guarantee key order. If the keys are passed in different order every time it runs, the md5 hash would be different even though the data may not be.

I'll give it a test and get back to you.

Reply:
You're right; although the keys aren't alphabetized they're consistently in the same order, and that's all that matters. Switched to `str(dictionary)`.

Reply:
👍 nice!